plugin/trino-prometheus/src/main/java/io/trino/plugin/prometheus/PrometheusClient.java

```diff
@@ -164,18 +164,15 @@ private Map<String, Object> fetchMetrics(JsonCodec<Map<String, Object>> metricsC
     public byte[] fetchUri(URI uri)
     {
         Request.Builder requestBuilder = new Request.Builder().url(uri.toString());
-        Response response;
```

Comment:
How did you found it? Can we have static code analyzer for it?

Reply:
okhttp reports leaked connection by logging a warning. I noticed "leaked" in the test output when trying to understand what's going on in https://github.com/trinodb/trino/issues/16933
it just happened that `TestMarkDistinctOperator` was running concurrently with `TestOAuth2WebUiAuthenticationFilterWithJwt`

i don't know about a way to automate that and refuse to do this as part of this PR, i hope this is ok